src/test/java/com/connexta/search/index/IndexComponentTest.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright (c) 2019 Connexta, LLC
+ *
+ * Released under the GNU Lesser General Public License version 3; see
+ * https://www.gnu.org/licenses/lgpl-3.0.html
+ */
+package com.connexta.search.index;
+
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+
+import com.connexta.search.common.exceptions.DetailedErrorAttributes;
+import com.connexta.search.index.configs.IndexControllerConfiguration;
+import com.connexta.search.index.controllers.IndexController;
+import com.connexta.search.index.controllers.IndexControllerTest;
+import com.connexta.search.rest.models.IndexRequest;
+import java.net.URI;
+import javax.inject.Inject;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.test.autoconfigure.web.reactive.WebFluxTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.reactive.server.WebTestClient;
+
+/**
+ * This class contains tests that use {@link
+ * org.springframework.test.web.reactive.server.WebTestClient} to test the {@link IndexController}
+ * that cannot be tested in {@link IndexControllerTest} and that do not interact with the {@link
+ * IndexService}.
+ */
+@WebFluxTest({
+  IndexController.class,
```

Comment:
❓ Should we be starting the ControllerAdvice and testing proper handling of error responses? I don't mind leaving a TODO as well.

Reply:
I actually need help with this. I want to verify that the InputStream contains the contents of the IRM, but I couldn't get my Matcher to work. Do you know why `StringUtils.equals(IOUtils.toString(argument, StandardCharsets.UTF_8), body)` doesn't work?

Reply:
@peterhuffer and I addressed this offline.